src/main/java/com/alexlasota/medicalclinic/service/PatientService.java

```diff
@@ -90,12 +89,24 @@ private void checkIfDataIsNotNull(Patient patient) {
         }
     }
 
-    private void checkIsEmailAvailable(Patient patient, String email) {
-        if (!patient.getEmail().equals(email) && patientRepository.getPatientByEmail(patient.getEmail()).isPresent()) {
+    private boolean checkIsEmailAvailable(Patient currentPatient, String email) {
```

Comment:
moglbys zrobic zeby metoda byla po prostu void i rzucala wyjatek jak zle a jak dobre to nic nie roibla po prostu